### ISSUE TITLE
Inconsistent function overload error with type and TypeVar

### ISSUE THREAD

**Issue by cdce8p** [microsoft/pylance-release#1407]
<h3> Environment data </h3>

-   Language Server version: 2021.6.0
-   OS and version: darwin x64
-   Python version (and distribution if applicable, e.g. Anaconda): 3.10.0b2
-   python.analysis.indexing: undefined
-   python.analysis.typeCheckingMode: basic

<h3>Code snippet</h3>

```py
from typing import overload, TypeVar

class A: ...
class B: ...

T = TypeVar("T", A, B)

@overload
def func(cls: type[A], var: int) -> A: ...

@overload
def func(cls: type[B], var: str) -> B: ...

def func(cls: type[T], var: int | str) -> T:   # error here
    reveal_type(cls)
    a = cls()
    reveal_type(a)
    return a
```

<h3> Expected behaviour </h3>

No error. This does seem to work with `mypy`. Am I missing something obvious?
```
test.py:15: note: Revealed type is "Type[test.A*]"
test.py:15: note: Revealed type is "Type[test.B*]"
test.py:17: note: Revealed type is "test.A"
test.py:17: note: Revealed type is "test.B"
```

<h3> Actual behaviour </h3>

```
Overloaded function implementation is not consistent with signature of overload 1
  Type "(cls: Type[T@func], var: int | str) -> T@func" cannot be assigned to type "(cls: Type[A], var: int) -> A"
    Parameter 1: type "Type[A]" cannot be assigned to type "Type[T@func]"
      Type "Type[A]" cannot be assigned to type "Type[T@func]"
        Type "Type[A]" cannot be assigned to type "A* | B*"
          "Type[type]" is incompatible with "Type[A]*"
          "Type[type]" is incompatible with "Type[B]*"PylancereportGeneralTypeIssues
```

<h3> Logs </h3>
--

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.6.1, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202161-9-june-2021
